### PR TITLE
Fixes #517

### LINKED ISSUE #517: Fatal (crash) bug: RAND
RAND only handles 32-bit requests. Cross over that barrier, and the bot fatally-crashes.

Edit: This should say: "the non-zero positive range of a signed 32-bit integer" (half-open interval: 0,2^31). Negatives and zero do not work. Floats are integer-truncated (floor?).

This shows-up when you try to create an arbitrarily high random number using the nice round number "10000000000" (10,000,000,000 or 1 + 10 zeroes), which is >= than 2^32.

[2018-05-15 17:07:29 -0800] \<Domino\> do tcl rand 2147483647
[2018-05-15 17:07:30 -0800] -Bot- [TCL (0ms)] 1552702703
[2018-05-15 17:07:31 -0800] \<Domino\> do tcl rand 2147483648
[2018-05-15 17:07:33 -0800] * Bot (~Bot@B4O86ED4.28DEC6G9.2975OOCE.IP) Quit (Client exited)

$ cat DEBUG
Debug (eggdrop v1.8.3) written Wed May 16 02:07:19 2018
Patch level: stable
Tcl library: /usr/share/tcltk/tcl8.6
Tcl version: 8.6.2 (header version 8.6.2)
Tcl is threaded
Compiled with IPv6 support
Compiled with TLS support
Configure flags: none
Compile flags: gcc -g -O2 -pipe -Wall -I.. -I..  -DHAVE_CONFIG_H -I/usr/include/tcl8.6
Link flags: gcc
Strip flags: touch
Context: tclhash.c/725, []
         tclhash.c/729, [Tcl proc: *raw:irc:msg, param:  $_raw1 $_raw2 $_raw3]
         tclhash.c/725, []
         tclhash.c/729, [Tcl proc: raw_commands, param:  $_pub1 $_pub2 $_pub3 $_pub4 $_pub5]
         tclhash.c/740, []
         tclhash.c/740, []
         tclhash.c/725, []
         tclhash.c/729, [Tcl proc: *raw:irc:msg, param:  $_raw1 $_raw2 $_raw3]
         tclhash.c/725, []
         tclhash.c/729, [Tcl proc: raw_commands, param:  $_pub1 $_pub2 $_pub3 $_pub4 $_pub5]
         tclhash.c/740, []
         tclhash.c/740, []
         tclhash.c/725, []
         tclhash.c/729, [Tcl proc: *raw:irc:msg, param:  $_raw1 $_raw2 $_raw3]
         tclhash.c/725, []
         tclhash.c/729 [Tcl proc: raw_commands, param:  $_pub1 $_pub2 $_pub3 $_pub4 $_pub5]

[Removed DCCSTAT table and interpretation/breakdown]

Shouldn't this just return a complaint, or process the value as 64-bit internally, just like EXPR does?

### PR DESCRIPTION
**Pull request by michaelortmann**
Found by: maimizuno
Patch by: michaelortmann
Fixes: #517

One-line summary:

Additional description (if needed):

Test cases demonstrating functionality (if applicable):

before:

.tcl rand 2147483647
Tcl: 1342848417
.tcl rand 2147483648
[05:02:10] * Last context: tclhash.c/729 [Tcl proc: *dcc:tcl, param:  $_dcc1 $_dcc2 $_dcc3]
[05:02:10] * Please REPORT this BUG!
[05:02:10] * Check doc/BUG-REPORT on how to do so.
[05:02:10] * Wrote DEBUG
[05:02:10] * SEGMENT VIOLATION -- CRASHING!
Connection closed by foreign host.

after:

.tcl rand 2147483647
Tcl: 1698643157
.tcl rand 2147483648
Tcl error: random limit must be equal to or less than 2147483647

### CODE REVIEW

**Comment by vanosg** [comment]
Thanks @michaelortmann !

@thommey : Do we need to / Does Tcl support numbers higher than 2^32 ?

Edit: Looked at source, we define RANDOM_MAX to be 2^31 - 1. For fun, should we up to ^64?

**Comment by michaelortmann** [comment]
regardless of tcl support for 64 bit:

if we dont change tcl_rand() in tcl.misc.c or its underlying function randint() then RANDOM_MAX must match the max output of randint(), which is 32bit as defined in eggdrop.h, regardless if the underlying function is random(), rand() or lrand48().

**Comment by vanosg** [comment]
Yep, spot on... we had some tangential conversation in the dev channel which is slowing us merging this is, but really its great. Thanks again for the submission... and feel free to hang out with us in #eggheads on Freenode if you want to chat or otherwise talk development with us, we really appreciate the support!

File: doc/sphinx_source/mainDocs/tcl-commands.rst [hunk 1]
```diff
@@ -2220,7 +2220,7 @@ myip
 rand <limit>
 ^^^^^^^^^^^^
 
-  Returns: a random integer between 0 and limit-1
+  Returns: a random integer between 0 and limit-1. limit must be greater than 0 and equal to or less than RAND_MAX, which is generally 2147483647. the underlying pseudo-random number generator is not cryptographically secure.
```
Comment (Cizzle) [inline doc/sphinx_source/mainDocs/tcl-commands.rst]: Please capitalise words starting a new sentence.

**Comment by Cizzle** [comment]
Thanks, LGTM.

**Pull request merged**